src/ceph-daemon/ceph-daemon

```diff
@@ -1616,6 +1616,105 @@ def command_rm_cluster():
+    cmd = find_program('firewall-cmd')
+    if not cmd:
+        logger.info('firewalld does not appear to be present')
+        return True  # no firewalld -> all good
+    (enabled, state) = check_unit('firewalld.service')
+    if not enabled:
+        logger.info('firewalld.service is not enabled')
+        return True
+    for svc in ['ceph', 'ceph-mon']:
+        out, err, ret = call([cmd, '--permanent', '--query-service', svc])
+        if ret:
+            if not fix:
+                logger.error('firewalld service %s is not enabled in current zone' % svc)
+                return False
+            logger.info('Enabling firewalld service %s in current zone...' % svc)
+            out, err, ret = call([cmd, '--permanent', '--add-service', svc])
+            if err:
+                logger.error('unabled to add service %s to current zone' % svc)
+                return False
+        else:
+            logger.info('firewalld service %s is enabled in current zone' % svc)
+    return True
+
+def command_check_host():
+    # caller already checked for docker/podman
+    logger.info('podman|docker (%s) is present' % container_path)
+
+    if not find_program('systemctl'):
+        raise RuntimeError('unable to location systemctl')
+    logger.info('systemctl is present')
+
+    if not find_program('lvcreate'):
+        raise RuntimeError('LVM does not appear to be installed')
+    logger.info('LVM2 is present')
+
+    # check for configured+running chronyd or ntp
+    if not check_time_sync():
+        raise RuntimeError('No time synchronization is active (checked all of %s)' %
+                           units)
+
+    # check for firewalld that will interfere with ceph daemons
+    if not check_firewalld():
+        raise RuntimeError('firewalld is present and ceph services are not enabled')
+
+    logger.info('Host looks OK')
+
+def command_prepare_host():
+    # if there is no systemd we're SOL
+    if not find_program('systemctl'):
+        raise RuntimeError('unable to location systemctl')
+
+    if not container_path:
+        logger.info('Trying to install podman or docker...')
+        out, err, ret = call(['dnf', 'install', '-y', 'podman'])
+        if err:
+            out, err, ret = call(['yum', 'install', '-y', 'podman'])
+        if err:
+            out, err, ret = call(['apt', 'install', '-y', 'podman'])
+        if err:
+            out, err, ret = call(['apt', 'install', '-y', 'docker.io'])
+        if err:
+            raise RuntimeError('unable to install podman|docker via dnf|yum|apt')
+
+    if not find_program('lvcreate'):
+        logger.info('Trying to install LVM2...')
+        out, err, ret = call(['dnf', 'install', '-y', 'lvm2'])
+        if err:
+            out, err, ret = call(['yum', 'install', '-y', 'lvm2'])
+        if err:
+            out, err, ret = call(['apt', 'install', '-y', 'lvm2'])
+        if err:
+            raise RuntimeError('unable to install lvm2 via dnf|yum|apt')
+
```

Comment:
as above the programs for check and prepare appear the same to me, so why not process a list. I'd also put the install logic itself in a separate function.

Reply:
Totally agree this is going to need to be smarter (soon).  I can drop the prepare part for this PR, or just merge this and continue to improve it.  I lean towards merging the half-baked thing just so that we can get the documented procedure in place sooner rather than later... and being able to say "run ./ceph-daemon prepare-host" as step 2 instead of "run apt install docker.io or zypper install podman or yum install podman or ..."...